### DISCUSSION TITLE
Nitty-Gritty: Get the ordering of the directed graph

### DISCUSSION THREAD

**Discussion by SortaASoraous** [mobxjs/mobx-state-tree#1942] [Q&A]
I know that MobX uses a directed graph of observables and node references, I'm curious, is it possible to get the "evaluation order" of  things like which nodes are evaluated first? 

Trying to build a visualization that shows the evaluation order of node references :)

**Comment by EmilTholin** [comment]
Hi @SortaASoraous!

Maybe the [MobX Introspection APIs](https://mobx.js.org/analyzing-reactivity.html#introspection-apis) could be of use here? Maybe `getDependencyTree` or `getObserverTree` could fit.
